widgetastic_manageiq/__init__.py

```diff
@@ -4051,3 +4051,461 @@ def data_for_timestamp(self, timestamp):
         for row in self.tooltip.rows():
             tooltip_data.update({row[0].text: row[1].text})
         return tooltip_data
+
+# Widget and its form views for Infra Planning Wizard
+
+
+class MultiSelectBox(Widget):
+    """Multi Select Box to be used in Infra Planning Wizard for v2v.
+
+    Args:
+        div_id (str): id of div element that contains multiselectbox
+    Usage:
+        e.g. target_clusters = MultiSelectBox('target_clusters')
+    """
+    ITEM_LOCATOR = ParametrizedLocator('.//div[contains(@id,{@div_id|quote})]/div/div/div/span'
+    '[contains(@class,"dual-pane-mapper-item-container")]')
+    SELECTED_ITEM_LOCATOR = ParametrizedLocator('.//div[contains(@id,{@div_id|quote})]/'
+        'div[contains(@class,"selected")]/div/div/span'
+    '[contains(@class,"dual-pane-mapper-item-container")]')
+
+    def __init__(self, parent, div_id, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.div_id = div_id
+
+    @property
+    def is_displayed(self):
+        script = "return document.getElementById('{}') !== null".format(self.div_id)
+        return self.browser.execute_script(script)
+
+    @property
+    def list_items(self):
+        """Returns list of all items."""
+        return [item.text for item in self.list_all_items_locators]
+
+    @property
+    def item_count(self):
+        """Returns item count."""
+        return len(self.list_items)
+
+    @property
+    def list_selected_items(self):
+        """Returns list of items that are currently selected."""
+        return [item.text for item in self.list_selected_items_locators]
+
+    @property
+    def list_all_items_locators(self):
+        """Returns list of item locators for each of the items."""
+        return self.browser.elements(self.ITEM_LOCATOR.locator)
+
+    @property
+    def list_selected_items_locators(self):
+        """Returns list of item locators for currently selected items."""
+        return self.browser.elements(self.SELECTED_ITEM_LOCATOR.locator)
+
+    def _get_all_options(self):
```

Comment:
**REQUIRED**
This should be a property.

Reply:
@kedark3 maybe a name change for `list_items` would help, like `all_items`, since we commonly use `all_<component>` for other widget methods.